aspnetcore/fundamentals/middleware/extensibility.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Middleware activation with nonconforming containers in ASP.NET Core
+author: guardrex
+description: Learn how to activate middleware with nonconforming containers in ASP.NET Core.
+ms.author: riande
+manager: wpickett
+ms.custom: mvc
+ms.date: 01/09/2018
+ms.topic: article
+ms.technology: aspnet
+ms.prod: asp.net-core
+uid: fundamentals/middleware/extensibility
+---
+# Middleware activation with nonconforming containers in ASP.NET Core
+
+By [Luke Latham](https://github.com/guardrex)
+
+[IMiddlewareFactory](/dotnet/api/microsoft.aspnetcore.http.imiddlewarefactory)/[IMiddleware](/dotnet/api/microsoft.aspnetcore.http.imiddleware) is an extensibility point for [middleware](xref:fundamentals/middleware/index) activation with nonconforming containers.
+
+`UseMiddleware` extension methods check if the middleware's registered type implements `IMiddleware`. If it does, the `IMiddlewareFactory` instance is used to resolve the `IMiddleware` implementation instead of using the convention-based middleware activation logic. The factory is registered as a scoped service with the app's service container.
+
+Benefits:
+
+* Automatic middleware activation with nonconforming containers
+* Strong typing of middleware
+* Activation per request (injection of scoped services)
```

Comment:
This is one of the main benefits and differences with the default middleware activation.

Reply:
Just move it to the top of the list? ... or would you like it to emphasize it in its own sentence/paragraph in the introduction?